### PR TITLE
Add spider/radar chart 

### PR DESCRIPTION
**Pull request by peter-stuart-turner**
Added a visualization: Spider/Radar chart (using D3) 

It has options to 'group by' selected columns, and compare the categories on a single metric for all of those columns. The comparisons are shown on the axis of the graph, each axis represents a category. 

There is also functionality to set the max value of the graph (outermost circle), this 'stretches' the axes on which the metric's values are plotted. 

I have not added any tests, as I am unsure of what I need to test, and how I need to do this - more than happy to add, just need some help there. 

Credit to Nadieh Bremer for the original JS and CSS, https://www.visualcinnamon.com 

(picture below)

screen shot 2018-04-24 at 1 45 53 pm

### CODE REVIEW

File: superset/viz.py [hunk 1]
```diff
@@ -2375,6 +2375,35 @@ def get_data(self, df):
...
+        metrics = fd.get('metrics')
+        groupby = fd.get('groupby')
+
+        axisMap = groupby[0]
+        valueMap = metrics[0].get('label')
+
+        created_data = []
+        for index, row in df.iterrows():
+            created_data.append({'axis': row[axisMap], 'value': row[valueMap]})
+        data = [created_data]
```
Comment (hughhhh) [inline superset/viz.py]: Why are you casting `created_data` in another list?

File: superset/viz.py [hunk 2]
```diff
@@ -2375,6 +2375,35 @@ def get_data(self, df):
...
+    def get_data(self, df):
+        fd = self.form_data
+
+        metrics = fd.get('metrics')
+        groupby = fd.get('groupby')
+
+        axisMap = groupby[0]
+        valueMap = metrics[0].get('label')
+
+        created_data = []
```
Comment (hughhhh) [inline superset/viz.py]: nit: `data = [{'axis': row[axisMap], 'value': row[valueMap]} for index, row in df.iterrows()]`

Comment (peter-stuart-turner) [reply superset/viz.py]: still learning python - thanks, will make the changes and add a unit test

File: superset/viz.py [hunk 3]
```diff
@@ -2375,6 +2375,35 @@ def get_data(self, df):
...
+    A spider / radar chart.
+    """
+
+    viz_type = 'spider_radar'
+    verbose_name = _('Spider Radar Diagram')
+    credits = (
+        '<a href=https://www.visualcinnamon.com/>Nadieh Bremer, Data Viz Expert</a>')
+    is_timeseries = False
+
+    def get_data(self, df):
```
[the remark this answers is not in the document]
Comment (peter-stuart-turner) [reply superset/viz.py]: yeah sure, will add one in soon

File: superset/viz.py (hunk 1 above)
Comment (peter-stuart-turner) [reply superset/viz.py]: To make room for later functionality of having multiple 'scenarios' pitted against one another:

	var data = [
					  [//iPhone
						{axis:"Battery Life",value:0.22},
						{axis:"Brand",value:0.28},		
					  ],[//Samsung
						{axis:"Battery Life",value:0.27},
						{axis:"Brand",value:0.16},
					  ],[//Nokia Smartphone
						{axis:"Battery Life",value:0.26},
						{axis:"Brand",value:0.10},
					  ]
					];
Example over here: 
https://www.visualcinnamon.com/2015/10/different-look-d3-radar-chart.html

File: superset/viz.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (peter-stuart-turner) [reply superset/viz.py]: Any chance I can get some help writing this unit test? I'm struggling to get it to run with tox - tox keeps failing in installing the dependencies. I'm runnning:

tox -e pylint -- tests/viz_tests.py

[the remark this answers is not in the document]
Comment (peter-stuart-turner) [reply superset/viz.py]: Also, is this what you meant? I mean, adding a test here. I can't seem to find where to put tests for the javascript code.

**Comment by mistercrunch** [comment]
Can you add an example in `data/__init__.py` this will also add test coverage on the python class. Also please rebase.

**Comment by peter-stuart-turner** [comment]
@mistercrunch 
in data/__init__.py, you mean load sample data and fit the spider as a slice in an example dashboard? Will need a bit of guidance through this first contrib to get comfortable adding other visualizations

File: superset/viz.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (hughhhh) [reply superset/viz.py]: I was specifically talking about python unit test. So to add test for these codeblocks you would reference this file.
https://github.com/hughhhh/incubator-superset/blob/master/tests/viz_tests.py

**Comment by hughhhh** [comment]
@peter-stuart-turner I think he's mean both correct me if I'm wrong @mistercrunch. I would use these PR for guidance on how to add examples.

https://github.com/apache/incubator-superset/pull/4242

**Comment by peter-stuart-turner** [comment]
@hughhhh @mistercrunch 

Hi guys, so I have written tests, as well as added the graph example to the 'miscelaneous' dashboard.

The checks are still failing. It seems to be because of one of the core tests (See below):

I really want to get this workflow cracked, as I suspect I'll be able to contribute a lot over the next few months/years - and will only get better. 

Would one of you mind helping me get this request merged? I am available to chat via skype,whatsapp, facebook etc. 

screen shot 2018-05-29 at 2 39 25 pm

**Comment by mistercrunch** [comment]
@peter-stuart-turner looks like you want to bump the assert statement to 5

**Comment by peter-stuart-turner** [comment]
@hughhhh 

Still failing. I'm not sure this is code I have written:

1. "ERROR: InvocationError for command '/home/travis/build/apache/incubator-superset/.tox/py36-postgres/bin/nosetests' (exited with code 1)
___________________________________ summary ____________________________________
ERROR:   py36-postgres: commands failed"
2. Same with sqlite ERROR:   py36-sqlite: commands failed
3. ERROR:   py27-mysql: commands failed

Any ideas? I really want to become a full time contributor on this project so want to get this process down.

**Comment by peter-stuart-turner** [comment]
@hughhhh

Still failing. It asks me to resolve conflicts in index.js? I do this (I think) and then I am faced with more errors below. 

 I'm not sure this is code I have written:

"ERROR: InvocationError for command '/home/travis/build/apache/incubator-superset/.tox/py36-postgres/bin/nosetests' (exited with code 1)
___________________________________ summary ____________________________________
ERROR: py36-postgres: commands failed"
Same with sqlite ERROR: py36-sqlite: commands failed
ERROR: py27-mysql: commands failed

Any ideas? I really want to become a full time contributor on this project so want to get this process down.

**Comment by mistercrunch** [comment]
Looks like you want to want to change that `4` to a `5` in `core_tests.py` line 367.

As far as fixing the merge conflict, one way to deal with it if you're confused is to copy the latest version of that file from master and re-write your change, which is just to add an entry for the spider chart in index.js.

**Comment by peter-stuart-turner** [comment]
@mistercrunch have managed to get them to pass!!:D

File: superset/assets/src/visualizations/spider_radar.js [hunk 4]
```diff
@@ -0,0 +1,410 @@
+import d3 from 'd3';
+import './spider_radar.css';
+
+function spiderRadarViz(slice, json) {
+  // console.log("Slice", slice);
```
Comment (hughhhh) [inline superset/assets/src/visualizations/spider_radar.js]: remove the comments here

File: superset/viz.py [hunk 5]
```diff
@@ -2457,6 +2457,38 @@ def get_data(self, df):
...
+        fd = self.form_data
+        metrics = fd.get('metrics')
+        groupby = fd.get('groupby')
+
+        axis_map = groupby[0]
+        # if not isinstance(metrics[0], str):
```
Comment (hughhhh) [inline superset/viz.py]: remove these comments

**Comment by hughhhh** [comment]
@peter-stuart-turner can you update this PR?

**Comment by peter-stuart-turner** [comment]
Will do it sometime this week or weekend

**Pull request closed without merging**